index.js

```diff
@@ -200,6 +200,10 @@ NYC.prototype.instrumentAllFiles = function (input, output, cb) {
   try {
     const stats = fs.lstatSync(input)
     if (stats.isDirectory()) {
+      if (output) {
+        fs.copySync(input, output)
```

Comment:
A potential issue with this approach:
```sh
nyc instrument --delete . self-coverage
```

This would result in copying the entire project into `./self-coverage`, which would be correct.  The issue is that `this.walkAllFiles` happens after these files are created.  I think we need to stop using `this.walkAllFiles`, instead perform:
```js
const files = this.exclude.globSync(input);
if (output) {
  fs.copySync(input, output);
}
files.forEach(relFile => {
  visitor(relFile);
})
```

This way `testExclude.globSync` is run before we create any files.  I know this shouldn't matter as someone following the above example should add `self-coverage` to the exclude list, but I think it will turn into an issue.

Reply:
Good point, `fs-extra.copySync` currently prevents this from happening as it won't let you copy `.` into `./dest`.  It seems a more custom solution without `fs-extra.copySync` may be a better approach.

Reply:
One other though, should the copy skip the `.git/` folder?

Reply:
Most likely, I can't think of a use case where you'd need to copy the `.git/` folder from `.` to `./dest` while instrumenting files.